modules/services/borgmatic.nix

```diff
@@ -0,0 +1,298 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.borgmatic;
+
+  mkNullableOption = args:
+    lib.mkOption (args // {
+      type = lib.types.nullOr args.type;
+      default = null;
+    });
+
+  mkRetentionOption = frequency:
+    mkNullableOption {
+      type = types.int;
+      description =
+        "Number of ${frequency} archives to keep. Use -1 for no limit.";
+      example = 3;
+    };
+
+  extraConfigOption = mkOption {
+    type = with types; attrsOf (oneOf [ str bool path int ]);
+    default = { };
+    description = "Extra settings.";
+  };
+
+  locationModule = types.submodule {
+    options = {
+      source_directories = mkOption {
```

Comment:
Since this is not using freeform modules, I think it would be more appropriate to use camel case for the options. I.e., `sourceDirectories` instead of `source_directories`.

Reply:
fixed

Reply:
Done. This change makes the implementation of `writeConfig` more complex as each attribute as to be referenced manually now.